### PR TITLE
Cdsid 1180

### PR DESCRIPTION
**Pull request by derekspalmer**

### CODE REVIEW

File: app/uk/gov/hmrc/cdsimportsddsfrontend/services/XmlWriter.scala [hunk 1]
```diff
@@ -44,4 +45,17 @@ object XmlWriterInstances {
     }
   }
 
+
+  implicit val additionalDocumentWriter: XmlWriter[AdditionalDocumentType] = new XmlWriter[AdditionalDocumentType] {
```
Comment (retroredge) [inline app/uk/gov/hmrc/cdsimportsddsfrontend/services/XmlWriter.scala:49]: This needs unit tests. Have a look at `XmlWriter_PreviousDocumentSpec`

**Review by retroredge — COMMENTED** [review]
Need to extend unit tests for Additional Doc in `DeclarationControllerSpec`

File: app/uk/gov/hmrc/cdsimportsddsfrontend/services/XmlWriter.scala (hunk 1 above)
Comment (derekspalmer) [reply app/uk/gov/hmrc/cdsimportsddsfrontend/services/XmlWriter.scala:49]: Tests added.

File: test/uk/gov/hmrc/cdsimportsddsfrontend/services/XmlWriter_AdditionalDocumentSpec.scala [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+  val additionalDocumentAll = AdditionalDocumentType(
+    categoryCode = Some("category_1"),
+    typeCode = Some("typeCode_1"),
+    id = Some("id_1"),
+    lpco = Some("lpco_1"),
+    name = Some("name_1")
+  )
+
+  val emptyFields = List(Some(""), None)
```
Comment (ellamdav) [inline test/uk/gov/hmrc/cdsimportsddsfrontend/services/XmlWriter_AdditionalDocumentSpec.scala:34]: 👍

**Review by ellamdav — APPROVED** [review]

**Pull request merged**